### ISSUE TITLE
[Feature] Gantt Series

### ISSUE THREAD

**Issue by Ovilia** [apache/echarts#19579]
### What problem does this feature solve?

An easier way to make charts like https://echarts.apache.org/examples/zh/editor.html?c=custom-gantt-flight

(More detail should be added later)

### What does the proposed API look like?

(More detail should be added later)

**Comment by Sadaf-A** [comment]
@Ovilia Greetings! I would liike to work on this

**Comment by Ovilia** [comment]
@Sadaf-A That's great! Gantt series is quite complicated and I'd like it to be similar to the [GitHub Roadmap](https://github.com/orgs/apache/projects/327/views/3). If you are interested, maybe we could start by discussing more details about expected features in the comments.

**Comment by Sadaf-A** [comment]
> @Sadaf-A That's great! Gantt series is quite complicated and I'd like it to be similar to the [GitHub Roadmap](https://github.com/orgs/apache/projects/327/views/3). If you are interested, maybe we could start by discussing more details about expected features in the comments.

yes, i would love to work on this can we start with what do you have in mind for this issue I would like to know a little more about our options and how to get started

**Comment by Ovilia** [comment]
@Sadaf-A Thanks for being interested. We really appreciate the help!

I don't have a clear mind about the product yet because I only did a rough research. Here's my current understanding.

The basic form should be probably similar to this [custom series example](https://echarts.apache.org/examples/en/editor.html?c=custom-gantt-flight) and [GitHub Roadmap page](https://github.com/orgs/apache/projects/327/views/3). Scrolling should be considered carefully because, to start with, as you can see from the custom series example, the default scrolling of dataZoom is not quite as expected with touchpad on MacOS. We shouldn't use DOM unless we cannot do it otherwise, to better support environments like Node.js or SVG. So I would suggest starting by improving current dataZoom component. You may try with it and see if there is anything needs to be enhanced.

I'm working on [Matrix Coordinate](https://github.com/apache/echarts/issues/19580), which may also be helpful to Gantt series with layouting. It's also not planned in details yet.

I would suggest researching about current Gantt series of other products and also the related issues in this project. And start by listing the useful features (you don't have to implement them all, just for brainstorming). Then, we could have further discussions based on this list.

Thanks!

**Comment by Ovilia** [comment]
After a full research, I found it too complicated (pretty much like a table with fixed header) to be implemented in ECharts so this will not be scheduled in v6.0. Perhaps we may provide this feature when we see some easier solutions.
